### ISSUE TITLE
UTF-8 is not correctly re-assembled

### ISSUE THREAD

**Issue by OrangeDog** [isaacs/sax-js#102]
When streaming UTF-8 data, the chunk boundaries can easily fall within a multi-byte sequence. These need to be re-assembled before attempting to convert back to a string.

Test code:
````javascript
var assert = require('assert'), saxStream = require('sax-js').createStream();
var b = new Buffer('误');

saxStream.on('text', function(text) {
    assert.equal(text, b.toString());
});

saxStream.write(new Buffer('<data>'));
saxStream.write(b.slice(0, 1));
saxStream.write(b.slice(1));
saxStream.write(new Buffer('</data>'));
````

Result:
AssertionError: "���" == "误"

**Comment by lovasoa** [comment]
Until this bug is fixed, you can use [utf8-stream](https://github.com/substack/utf8-stream).
I think sax should use it itself, but I'm not the maintainer would like to add a dependency to *sax*. 

The work-around:
```javascript
var assert = require('assert'),
      saxStream = require('sax-js').createStream();

inStream = require("utf8-stream")()
inStream.pipe(saxStream);

var b = new Buffer('误');

saxStream.on('text', function(text) {
    assert.equal(text, b.toString());
});

inStream.write(new Buffer('<data>'));
inStream.write(b.slice(0, 1));
inStream.write(b.slice(1));
inStream.write(new Buffer('</data>'));
```

**Comment by OrangeDog** [comment]
@isaacs has merged the fix in 0.5.5 afaict
